__tests__/getProducts.spec.js

```diff
@@ -26,33 +26,117 @@ describe("User story #2 retrieves products from FaunaDB then returns it as an HT
     expect(caught).toEqual(3);
   });
 
-  // Response Data
-  let response = null;
+  
+  // Page 1 of shop
+  var response = null;
   test("The get request works and returns a valid response.", async () => {
     response = await axios.get("http://localhost:9000/getProducts");
     expect(response).not.toEqual(null);
     expect(response.status).toEqual(200);
     expect(response.data).not.toEqual(null);
   });
 
-  // Test pagination once it's available
+  test("The number of products should be less than or equal to 8.", async () => {
+    expect(response.data.data.length <= 8);
+    console.log(response.data.data.length)
+  });
 
-  test("Should retrieve the correct details (preview image, product name, product price, collection, product category) for each product from the database.", () => {
-    expect(response.data).not.toEqual(null);
+  test("Should retrieve the correct details (preview image, product name, product price, before cursor, after cursor) for each product from the database.", () => {
+    expect(response.data.data).not.toEqual(null);
 
     // Test every element to see if it is a valid JS object that we are expecting
-    response.data.forEach((product) => {
+    response.data.data.forEach((product) => {
       expect(product).toEqual(
         expect.objectContaining({
           _id: expect.any(String),
+          slug: expect.any(String),
           name: expect.any(String),
           price: expect.any(Number),
           mainImage: expect.any(String),
         })
       );
+      // expect before and after cursor
       expect(product.mainImage).toMatch(
         /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/
       );
     });
+
+    // Must contain the after & before field, can be null or String
+    expect(response.data.after).toBeDefined();
+    expect(response.data.before).toBeDefined();
+    
+  });
+
+  // Pagination test: next page and check EVERY page
+  // Tests from first page to last page
+  test("Should successfully retrieve the NEXT page of products successfully", async () => {
+    // Next page available
+    while (response.data.after !== null){
+      //cursor = response.data.after
+      response = await axios.get(`http://localhost:9000/getProducts?cursor=${response.data.after}`);
+      
+      // Number of products <= 8
+      expect(response.data.data.length <= 8);
```

Comment:
Sorry, not too familiar yet with Jest's syntax. Does `expect(response.data.data.length <= 8)` work the same as `expect(response.data.data.length).toEqual(8)`?

Reply:
ah nono, the prior's literally just a js expression, it's possible kasi that a page might not be fully populated depending on (1) where the cursor is and (2) the number of products left, hence the less than condition

tho Jest's `.toEqual()` function is equivalent to Java String's `.equals` function, it checks the value rather than the reference

Reply:
Ah, alright. So it does return a boolean value and I'm guessing that `expect()` is expecting a `true`.